src/code-templates/services/order-service/data-access-prisma/order-repository.ts

```diff
@@ -0,0 +1,50 @@
+import * as prismaTypes from './prisma/client/index.d';
+// eslint-disable-next-line import/no-relative-packages
+import { PrismaClient } from './prisma/client';
+
+const prisma = new PrismaClient();
+
+// ️️️✅ Best Practice: The repository pattern - This is a plain JS object (POJO) that is returned to the domain layer
```

Comment:
I don't know if it's best practice, for small application it add unnecessary layer and for complex and big application it's needed but we aim to split when we reach enough complexity

Another complexity is maintaining multiple interfaces

Reply:
I find the effort of adding one more class to be negligible comparing with exposing the DAL narratives to the domain